cmd/kubeadm/app/phases/controlplane/users_linux.go

```diff
@@ -0,0 +1,260 @@
+	cmd.SetStderr(&buff)
+	if err := cmd.Run(); err != nil {
+		return errors.New(fmt.Sprintf("failed to add group %s: %s : %s", groupname, err.Error(), buff.String()))
+	}
+	return nil
+}
+
+// createControlPlanUser creates a user. It does not check if the user
+// exists before creating it.
+func createControlPlanUser(username string) error {
+	exec := exec.New()
+	cmd := exec.Command("useradd", "-r", "-s", "/bin/false", username)
+	buff := bytes.Buffer{}
+	cmd.SetStderr(&buff)
+	if err := cmd.Run(); err != nil {
+		return errors.New(fmt.Sprintf("failed to add user %s: %s : %s", username, err.Error(), buff.String()))
+	}
+	return nil
+}
+
+func getOrCreateControlPlanGroup(groupname string) (int64, error) {
+	gid, found, err := getControlPlaneGroupId(groupname)
+	if err != nil {
+		return 0, err
+	}
+
+	if found {
+		return gid, nil
+	}
+
+	if err := createControlPlanGroup(groupname); err != nil {
+		return 0, err
+	}
+
+	gid, _, err = getControlPlaneGroupId(groupname)
+	if err != nil {
+		return 0, err
+	}
+
+	return gid, nil
+}
+
+func getOrCreateControlPlanUser(username string) (int64, error) {
+	uid, found, err := getControlPlaneUserId(username)
+	if err != nil {
+		return 0, err
+	}
+
+	if found {
+		return uid, nil
+	}
+
+	if err := createControlPlanUser(username); err != nil {
+		return 0, err
+	}
+
+	uid, _, err = getControlPlaneUserId(username)
+	if err != nil {
+		return 0, err
+	}
+
+	return uid, nil
+}
+
+func removeUser(username string) error {
+	exec := exec.New()
+	cmd := exec.Command("userdel", username)
+	buff := bytes.Buffer{}
+	cmd.SetStderr(&buff)
+	if err := cmd.Run(); err != nil {
+		return errors.New(fmt.Sprintf("failed to remove user %s: %s : %s", username, err.Error(), buff.String()))
+	}
+	return nil
+}
+
+func removeGroup(groupname string) error {
+	exec := exec.New()
+	cmd := exec.Command("groupdel", groupname)
+	buff := bytes.Buffer{}
+	cmd.SetStderr(&buff)
+	if err := cmd.Run(); err != nil {
+		return errors.New(fmt.Sprintf("failed to remove group %s: %s : %s", groupname, err.Error(), buff.String()))
+	}
+	return nil
+}
+
+var RemoveUsersAndGroupsFn = func() error {
+	users := []string{
+		kubeadmconstants.KubeSchedulerUserName,
+		kubeadmconstants.KubeControllerManagerUsername,
+		kubeadmconstants.KubeAPIServerUserName,
+		kubeadmconstants.EtcdUserName,
+	}
+	groups := []string{kubeadmconstants.ServiceAccountKeyReadersGroupName}
+	for _, user := range users {
+		_, found, err := getControlPlaneUserId(user)
+		if err != nil {
+			return err
+		}
+		if found {
+			if err := removeUser(user); err != nil {
+				return err
+			}
+		}
+	}
+	for _, group := range groups {
+		_, found, err := getControlPlaneGroupId(group)
+		if err != nil {
+			return err
+		}
+		if found {
+			if err := removeGroup(group); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func RemoveUsersAndGroups() error {
+	return RemoveUsersAndGroupsFn()
+}
+
+var GetUserAndGroupIDsFn = func() (map[string]int64, error) {
```

Comment:
This needs a comment and warning that this should not be called directly and is only for testing to override the function behavior for tests.

Also I hate that I have to do this and am open to suggestions on doing it another way if possible.